### PR TITLE
Clear inactive assignee for old high-priority bugs without needinfo

### LINKED ISSUE #1759: When a bug is high priority but the priority was set earlier than a year ago and the assignee is inactive, clear the assignee without needinfoing the triage owner

### PR DESCRIPTION
**Pull request by benjaminmah**
Resolves #1759.

Depends on #2515 to remove the TypeError.

## Checklist

- [ ] Type annotations added to new functions
- [ ] Docs added to functions touched in main classes
- [ ] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/bugbot/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

**Comment by suhaibmujahid** [comment]
Could you please rebase?

File: bugbot/rules/assignee_no_login.py [hunk 1]
```diff
@@ -68,6 +78,34 @@ def get_bugs(self, *args, **kwargs):
...
+                    )
+
+            bug_histories[bug_id] = history_entries
+
+        Bugzilla(
+            bugids=bug_ids,
+            historyhandler=history_handler,
+        ).get_data().wait()
+
+        return bug_histories
```
Comment (suhaibmujahid) [inline bugbot/rules/assignee_no_login.py:107]: Why do we need to retrieve the history in a separate query?

Comment (benjaminmah) [reply bugbot/rules/assignee_no_login.py]: I realized I wasn't populating the `data` dictionary with the `history`, but I've changed that here and removed the separate history query: 6b2ea7f3d504fd06944cb56bcd554906d04388da

File: bugbot/rules/assignee_no_login.py [hunk 2]
```diff
@@ -142,6 +165,7 @@ def handle_bug(self, bug, data):
             "priority": bug["priority"],
             "severity": bug["severity"],
             "keywords": bug["keywords"],
+            "history": bug["history"],
```
Comment (suhaibmujahid) [inline bugbot/rules/assignee_no_login.py:168]: Why do we need to keep the history? Could we use it here directly and pass the result instead?

Comment (benjaminmah) [reply bugbot/rules/assignee_no_login.py:168]: Yes we could, but then I would have overwrite/add on top of the functions `bughandler` and `get_bugs`. Is there a major benefit of changing the number of outputs of `handle_bug`?

Comment (suhaibmujahid) [reply bugbot/rules/assignee_no_login.py:168]: I meant to use the `get_priority_change_date()` here and pass the "priority" value instead of the whole history.

Comment (suhaibmujahid) [reply bugbot/rules/assignee_no_login.py:168]: I meant to use the `get_priority_change_date()` here and pass `is_old_priority` instead of the whole history.

Comment (benjaminmah) [reply bugbot/rules/assignee_no_login.py]: Got it! Like this? 623e88aa27cc8193e1416112a82cf65f5aafac81

File: bugbot/rules/assignee_no_login.py [hunk 3]
```diff
@@ -60,6 +62,7 @@ def get_max_actions(self):
 
     def get_bugs(self, *args, **kwargs):
         bugs = super().get_bugs(*args, **kwargs)
+
```
Comment (suhaibmujahid) [inline bugbot/rules/assignee_no_login.py:65]: Unrelated change.

File: bugbot/rules/assignee_no_login.py [hunk 4]
```diff
@@ -100,13 +103,19 @@ def add_action(self, bug):
         default_assignee = self.default_assignees[prod][comp]
         autofix = {"assigned_to": default_assignee}
 
+        is_old_priority = bug["is_old_priority"]
+
```
Comment (suhaibmujahid) [inline bugbot/rules/assignee_no_login.py:107]: Why do we need to assign it to a variable here?

Comment (benjaminmah) [reply bugbot/rules/assignee_no_login.py]: We don't really need it, removed it here: 737ccb2579dd034fc3c7a0ad12004fbdce6fe8aa

File: bugbot/rules/assignee_no_login.py (hunk 3 above)
Comment (benjaminmah) [reply bugbot/rules/assignee_no_login.py]: Removed here: 737ccb2579dd034fc3c7a0ad12004fbdce6fe8aa
